### PR TITLE
Modernize documentation using mkdocs

### PR DESCRIPTION
**Pull request by farodin91**
Mailing list discussion: https://groups.google.com/forum/#!topic/janusgraph-dev/KzyxrrPfnek

TODO
* [ ] travis integration
* [ ] multi version support
* [ ] check that documentation is include in released zip of janusgraph
* [x] check links
* [x] link checker integration (by mkdocs only for md)
* [x] Make release and changelog more visible
* [x] Add links to old documtation into the changelog
* [ ] Check all old reference to asciidoc and remove them
* [ ] Rebase to branch 0.2 (Only 4 commit more than in current master) **We would have all currently support version as modern documentation.** 

Can be done in follow up PR:
* fix all code box
* spellcheck
* convert documentation into a pdf
* automate reading of current version maven (also deps)
* documentation only travis ci builds

Demo: https://farodin91.github.io/janusgraph
-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by FlorianHockmann** [comment]
janusgraph_change_all_the_lines

😱

**Comment by farodin91** [comment]
:see_no_evil:

**Comment by porunov** [comment]
@farodin91 Thank you for working on this! I really like a new design)

I see that there is a message in the footer:
`powered by MkDocs and Material for MkDocs`

Can you remove it? Looks like this issue can help: https://github.com/squidfunk/mkdocs-material/issues/373

Also, I noticed that some code parts are grey only and some use different colors. I didn't get why yet.
For example if we go here:
https://farodin91.github.io/janusgraph/intro/

This part of code is grey only:
1a
And this one uses colors:
1b

Do you know why it happens? Looks like codes are mostly identical but one of them don't use colors .

**Comment by farodin91** [comment]
@porunov footer: why you want that we remove which design we are using? The old documentation has already such footer. 
Code boxes: you have to add language to all code box in markdown which take a lot of time.

**Comment by porunov** [comment]
@farodin91 Maybe it is just my personal preference of choice but I always didn't like such footers. I don't know why but this information distracts me when I see it on the main page. I don't mind creating a separate documentation page where we describe which design and generator we are using and other related information. It's not a requirement, it is just a suggestion. If you wish you can keep that information. I don't mind.

**Comment by FlorianHockmann** [comment]
I think we need the footer in general for the copyright notice and for all the trademarks.
Whether it needs to contain a reference to _MkDocs_ may be up for discussion. I personally think that it's good practice to give them attribution if we use that tool to generate our docs. It makes it easier for others who like our docs to find out how they are generated so they can also use _MkDocs_ for their own projects if they want to.

> Code boxes: you have to add language to all code box in markdown which take a lot of time.

@farodin91: Could you add things like these that aren't done yet to your TODO list in the PR description? That way reviewers know what they can ignore for now. A task list would probably be a good idea as it allows you to tick off items as you fix them.

**Comment by porunov** [comment]
> I think we need the footer in general for the copyright notice and for all the trademarks.
> Whether it needs to contain a reference to _MkDocs_ may be up for discussion. I personally think that it's good practice to give them attribution if we use that tool to generate our docs. It makes it easier for others who like our docs to find out how they are generated so they can also use _MkDocs_ for their own projects if they want to.

@FlorianHockmann As I said, it is just my personal preference of choice. I would create a separate documentation page to reference all used tools like: docs generator, theme which is used, CLA bot (actually I still don't know which one is used) and other things. As for me, all this information in the footer is perceived as a distraction. Possibly I just don't like big footers, but again, I don't mind if that information will be placed there.

**Comment by farodin91** [comment]
@FlorianHockmann  could you merge?

**Comment by FlorianHockmann** [comment]
@farodin91 Please sign your commit first and maybe reword the commit message. We might squash commits later on the feature branch, but we could also decide to keep them as is, especially when they come from different contributors.
BTW: The preview is currently served from your fork. Any idea on how we could show an updated preview from the feature branch? gh-pages could be problematic as it probably serves our productive docs right now.

**Comment by farodin91** [comment]
@FlorianHockmann I will sign the commit
Preview: My idea is to use gh-pages in this repo, because this branch is not used currently https://janusgraph.github.io/janusgraph/. If my works correctly, i will start work on travis ci and this will lead to an automatic deployment to https://janusgraph.github.io/janusgraph/. So, we can use the branch as a test branch.

**Comment by farodin91** [comment]
@FlorianHockmann Improved the wording of the commit
